daemon/containerd/image.go

```diff
@@ -256,6 +256,24 @@ func (i *ImageService) resolveImage(ctx context.Context, refOrID string) (contai
 			return containerdimages.Image{}, images.ErrImageDoesNotExist{Ref: parsed}
 		}
 
+		// If reference is both Named and Digested, make sure we don't match
+		// images with a different repository even if digest matches.
+		// For example, busybox@sha256:abcdef..., shouldn't match asdf@sha256:abcdef...
+		if parsedNamed, ok := parsed.(reference.Named); ok {
```

Comment:
You could try and add a `"name=="+parsedName.Name()` above on line 251 in the case where it's also a named reference

Reply:
It won't work, the filters in c8d image list are OR'd not AND'd.